pkg/rancher-desktop/components/StatusBarItem.vue

```diff
@@ -0,0 +1,77 @@
+<script lang="ts">
+import Vue, { PropType } from 'vue';
+export default Vue.extend({
+  props: {
+    data: {
+      type:    Object as PropType<{ label: string, value: string }>,
+      default: null,
+    },
+    icon: {
+      type:     String,
+      required: true,
+    },
+    showIcon: {
+      type:    Boolean,
+      default: false,
+    },
+    showText: {
+      type:    Boolean,
+      default: true,
+    },
+    hasComponent: {
+      type:    Boolean,
+      default: false,
+    },
+  },
+  computed: {
+    getText(): string {
+      if (this.showText && this.showIcon) {
+        return this.data.value;
+      }
+
+      return this.getFullText;
+    },
+    getFullText(): string {
+      return `<b>${ this.t(this.data.label) }</b>: ${ this.data.value }`;
+    },
+    getTooltip(): { content: string, placement: string, classes: string } {
+      return {
+        content:   this.getFullText,
+        placement: 'left',
+        classes:   'tooltip-footer',
+      };
+    },
+  },
+});
+</script>
+
+<template>
+  <div
+    v-if="hasComponent || data"
+    class="item"
+  >
+    <span
+      v-if="!hasComponent"
+      v-tooltip="getTooltip"
+    >
+      <i
+        v-if="showIcon"
+        class="item-icon"
+        :class="icon"
+      />
+      <span
+        v-if="showText"
+        v-html="getText"
+      >
+      </span>
+    </span>
+    <slot v-else name="content"></slot>
```

Comment:
I'm of the mind that slots should be exposed by default inside of a component, otherwise we end up in a scenario where some magic internal component logic can disable a slot unexpectedly.